File: 915-partition-array-into-disjoint-intervals/915-partition-array-into-disjoint-intervals.cpp
class Solution {
public:
    int partitionDisjoint(vector<int>& nums) {
        int maxUntilNow = nums[0], leftPartMax = nums[0], partIdx = 0;
        for(int i = 1; i < nums.size(); ++i) {
            maxUntilNow = max(maxUntilNow, nums[i]);
            if(nums[i] < leftPartMax) {
                leftPartMax = maxUntilNow;
                partIdx = i;
            }
        }
        return partIdx + 1;
    }
};

File: 915-partition-array-into-disjoint-intervals/915-partition-array-into-disjoint-intervals.py
class Solution:
    def partitionDisjoint(self, nums: List[int]) -> int:
        maxUntilNow, leftPartMax, partIdx = nums[0], nums[0], 0
        for i in range (len(nums)):
            maxUntilNow = max(maxUntilNow, nums[i])
            if nums[i] < leftPartMax:
                leftPartMax, partIdx = maxUntilNow, i
        return partIdx + 1
                